src/resources/touch_esp32.cc

```diff
@@ -274,13 +272,12 @@ PRIMITIVE(read) {
   ARGS(IntResource, resource);
   touch_pad_t pad = static_cast<touch_pad_t>(resource->id());
 
-#if CONFIG_IDF_TARGET_ESP32S3 || CONFIG_IDF_TARGET_ESP32S2
-  uint32_t val;
-  esp_err_t err = touch_pad_read_raw_data(pad, &val);
-#else
+#ifdef CONFIG_IDF_TARGET_ESP32
   uint16_t val;
-  esp_err_t err = touch_pad_read(pad, &val);
+#else
+  uint32_t val;
```

Comment:
C3 used to use uint16, no?

Reply:
Only ESP32/S2/S3 supports touch pad sensor, and data type is uint32_t.

Reply:
The C3 is already excluded by the big #ifdef at the top.